subrecipes-mcp/main.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import asyncio
+import json
+from typing import Awaitable, Callable
+from pathlib import Path
+from fastmcp import FastMCP, Context
+from mcp import types
+
+server = FastMCP(
+    "Subrecipes 🚀",
+    instructions="This extension allows you to run subrecipes.",
+)
+
+del server._mcp_server.request_handlers[
+    types.ListResourcesRequest
+]  # Otherwise the Goose system prompt says this supports resources
+
+
+def runner(subrecipe: dict) -> tuple[str, Callable[[Context], Awaitable[str]]]:
```

Comment:
@lifeizhou-ap my thinking is that we turn these into tools added and handled by Goose itself, without the MCP layer at all. Then we can start with an implementation where Goose still shells out to itself. And that leaves the door open to Goose running the subrecipe in its own separate agent in-process